stdlib/digraph.mc

```diff
@@ -153,6 +160,15 @@ let digraphAddVertex = lam v. lam g.
 let digraphMaybeAddVertex = lam v. lam g.
   digraphAddVertexCheck v g false
 
+-- Add a vertex v to g. Updates the vertex v in graph g if v already exists in g.
+let digraphAddUpdateVertex = lam v. lam g : Digraph v l.
+  if digraphHasVertex v g then
+    let edgeList = mapLookupOrElse (lam. error "Lookup failed") v g.adj in
+    let m = mapRemove v g.adj in
+    {g with adj = mapInsert v edgeList m}
```

Comment:
Maybe I'm missing something, but wouldn't it be simpler with just:
```suggestion
  if digraphHasVertex v g then g
```

Reply:
If the vertex v exists, I still want to update the v in the graph (maybe same ident [lets say comps looking to ident] but non-compared parts within v might be updated). Is the motivation clear now?

Reply:
Ah, I see!